sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/EventProcessorClient.cs

```diff
@@ -919,28 +927,30 @@ protected override Task UpdateCheckpointAsync(string partitionId,
             cancellationToken.ThrowIfCancellationRequested<TaskCanceledException>();
             Argument.AssertNotNull(partitionId, nameof(partitionId));
 
-            Logger.UpdateCheckpointStart(partitionId, Identifier, EventHubName, ConsumerGroup);
+            var offsetString = offset.ToString(CultureInfo.InvariantCulture);
+            Logger.UpdateCheckpointStart(partitionId, Identifier, EventHubName, ConsumerGroup, sequenceNumber?.ToString(CultureInfo.InvariantCulture), offsetString);
 
             using var scope = ClientDiagnostics.CreateScope(DiagnosticProperty.EventProcessorCheckpointActivityName, ActivityKind.Internal);
             scope.Start();
 
             try
             {
-                return CheckpointStore.UpdateCheckpointAsync(FullyQualifiedNamespace, EventHubName, ConsumerGroup, partitionId, offset, sequenceNumber, cancellationToken);
+                var checkpointPosition = new CheckpointPosition(offsetString, sequenceNumber ?? long.MinValue);
```

Comment:
Not sure that I'm following.   Offset in the service is a string value, so there's no numeric parsing on that side.   Everything on the client side should be fully moved over to string-based and also shouldn't be parsing.    Can you help me understand what path I may be overlooking?

Reply:
I was thinking about the GeoDR in-place upgrade case, but upon further thought I don't think this case would ever get to this point because accessing the offset value would throw an exception. I remember when I was working on this I had to carefully preserve functionality on the numeric offset path, but with the new architect guidance there's no need to worry about that.

Reply:
I think that I'm following now.  There's no harm in the offset being numeric or legacy format when creating the checkpoint.  If that happens and we hit a failover, then the service is going to return a unique error condition that we can use to identify the case and ignore the checkpoint.   If we throw here or we hit an error after failover, the outcome is the same - we'll have no valid checkpoint and will fall back to the default position.

That make sense?

Reply:
Yeah that makes sense. I think it makes sense to solve this with a combination of service side errors / exceptions in other places to catch things earlier (like in the case of accessing the offset property).